Eos/GammaLaw/EOS.H

```diff
@@ -9,278 +9,324 @@
+  AMREX_FORCE_INLINE
+  void PYT2RE(
+    amrex::Real P,
+    amrex::Real Y[],
+    amrex::Real T,
+    amrex::Real& R,
+    amrex::Real& E)
+  {
+    amrex::Real wbar;
+    Y2WBAR(Y, wbar);
+    R = P * wbar / (Constants::RU * T);
+    E = P / (R * (gamma - 1.0));
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void PYT2R(amrex::Real P, amrex::Real Y[], amrex::Real T, amrex::Real& R)
+  {
+    amrex::Real wbar;
+    Y2WBAR(Y, wbar);
+    R = P * wbar / (Constants::RU * T);
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void RYP2E(amrex::Real R, amrex::Real Y[], amrex::Real P, amrex::Real& E)
+  {
+    amrex::Real wbar;
+    Y2WBAR(Y, wbar);
+    E = P / (R * (gamma - 1.0));
+  }
 
-} // namespace EOS
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void TY2E(amrex::Real T, amrex::Real Y[], amrex::Real& E)
+  {
+    amrex::Real ei[NUM_SPECIES];
+    T2Ei(T, ei);
+    E = 0.0;
+    for (int n = 0; n < NUM_SPECIES; n++)
+      E += Y[n] * ei[n];
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void RTY2E(amrex::Real /*R*/, amrex::Real T, amrex::Real Y[], amrex::Real& E)
+  {
+    TY2E(T, Y, E);
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void T2Hi(amrex::Real T, amrex::Real Hi[])
+  {
+    amrex::Real wbar = Constants::AIRMW;
+    const amrex::Real Cv = Constants::RU / (wbar * (gamma - 1.0));
+    for (int n = 0; n < NUM_SPECIES; n++)
+      Hi[n] = Cv * T * gamma;
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void S(amrex::Real& s) { s = 1.0; }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void TY2G(amrex::Real /*T*/, amrex::Real* /*Y*/, amrex::Real& G)
+  {
+    G = gamma;
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void RTY2G(amrex::Real /*R*/, amrex::Real T, amrex::Real* Y, amrex::Real& G)
+  {
+    TY2G(T, Y, G);
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void TY2H(amrex::Real T, amrex::Real Y[NUM_SPECIES], amrex::Real& H)
+  {
+    amrex::Real Hi[NUM_SPECIES];
+    amrex::Real wbar = Constants::AIRMW;
+    const amrex::Real Cv = Constants::RU / (wbar * (gamma - 1.0));
+    H = 0.0;
+    for (int n = 0; n < NUM_SPECIES; n++) {
+      Hi[n] = Cv * T * gamma;
+      H = H + Y[n] * Hi[n];
+    }
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void RPE2dpdr_e(
+    amrex::Real /*R*/, amrex::Real /*P*/, amrex::Real E, amrex::Real& dpdr_e)
+  {
+    dpdr_e = (gamma - 1.0) * E;
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void RG2dpde(amrex::Real R, amrex::Real /*G*/, amrex::Real& dpde)
+  {
+    dpde = (gamma - 1.0) * R;
+  }
+
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void RTY2dpde_dpdre(
+    amrex::Real R,
+    amrex::Real T,
+    amrex::Real Y[],
+    amrex::Real& dpde,
+    amrex::Real& dpdr_e)
+  {
+    amrex::Real G, P;
+    TY2G(T, Y, G);
+    RTY2P(R, T, Y, P);
+    dpde = (G - 1.0) * R;
+    dpdr_e = P / R;
+  }
+
+  template <class... Args>
+  AMREX_GPU_HOST_DEVICE GammaLaw(Args...)
+  {
+  }
+  amrex::Real gamma{1.4};
```

Comment:
Would it make sense to specify this in `PhysicsConstants.H` instead?

Reply:
done